src/Symfony/Component/Workflow/EventListener/GuardListener.php

```diff
@@ -28,15 +29,17 @@ class GuardListener
     private $authenticationChecker;
     private $trustResolver;
     private $roleHierarchy;
+    private $validator;
 
-    public function __construct($configuration, ExpressionLanguage $expressionLanguage, TokenStorageInterface $tokenStorage, AuthorizationCheckerInterface $authenticationChecker, AuthenticationTrustResolverInterface $trustResolver, RoleHierarchyInterface $roleHierarchy = null)
+    public function __construct($configuration, ExpressionLanguage $expressionLanguage, TokenStorageInterface $tokenStorage, AuthorizationCheckerInterface $authenticationChecker, AuthenticationTrustResolverInterface $trustResolver, RoleHierarchyInterface $roleHierarchy = null, ValidatorInterface $validator = null)
```

Comment:
looks okay to me, but let's wait for @lyrixx and the other @symfony/deciders to share their opinion before spending time on this